x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutor.java

```diff
@@ -51,8 +90,17 @@ public void runPolicy(String policyId, ActionListener<PolicyExecutionResult> lis
     }
 
     public void runPolicy(String policyName, EnrichPolicy policy, ActionListener<PolicyExecutionResult> listener) {
-        EnrichPolicyRunner runnable = new EnrichPolicyRunner(policyName, policy, listener, clusterService, client,
-            indexNameExpressionResolver, nowSupplier, fetchSize);
-        threadPool.executor(ThreadPool.Names.GENERIC).execute(runnable);
+        if (tryLockingPolicy(policyName)) {
+            try {
+                Runnable runnable = createPolicyRunner(policyName, policy, new PolicyUnlockingListener(policyName, listener));
+                threadPool.executor(ThreadPool.Names.GENERIC).execute(runnable);
+            } catch (Exception e) {
+                // Be sure to unlock if submission failed.
+                releasePolicy(policyName);
+                throw e;
+            }
+        } else {
+            throw new ElasticsearchException("Policy execution failed. Policy execution for [{}] is already in progress.", policyName);
```

Comment:
What does this map to in the HTTP response ?  (should this be 429 ?)

Reply:
429 would be a good response code for this. I'll switch this to `EsRejectedExecutionException`